app/knight.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from app.ammunition.armour import Armour
+from app.ammunition.potion import Potion
+from app.ammunition.weapon import Weapon
+
+
+class Knight:
+    _all_knights = []
+
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: [Armour],
+            weapon: Weapon,
+            potion: Potion,
+    ) -> None:
+
+        self._name = name
+        self._hp = hp
+        self._power = power
+        self._protection = 0
+        self._weapon = weapon
+        self._armour = armour
+        self._potion = potion
+        self.prepare_for_battle()
+
+        Knight._all_knights.append(self)
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def power(self) -> int:
+        return self._power
+
+    @property
+    def hp(self) -> int:
+        return self._hp
+
+    @property
+    def protection(self) -> int:
+        return self._protection
+
+    @property
+    def armour(self) -> [Armour]:
+        return self._armour
+
+    @property
+    def weapon(self) -> Weapon:
+        return self._weapon
+
+    @property
+    def potion(self) -> [Potion]:
+        return self._potion
+
+    def _apply_armour(self) -> None:
+        self._protection += sum([armour.protection
+                                for armour in self._armour])
+
+    def _apply_potion(self) -> None:
+        effects = self._potion.effect
+        # print(effects)
+        if "power" in effects:
+            self._power += effects["power"]
+        if "hp" in effects:
+            self._hp += effects["hp"]
+        if "protection" in effects:
+            self._protection += effects["protection"]
```

Comment:
Read the checklist:

![image](https://user-images.githubusercontent.com/104218550/219120210-95b7db14-5e8e-4f79-8814-1eb244f4fc2b.png)


Hint: use `setattr` for this purpose